### PR TITLE
Port tests in projection.rs to sqllogictest

### LINKED ISSUE #8210: Port tests in `projection.rs` to sqllogictest
### Is your feature request related to a problem or challenge?

Part of https://github.com/apache/arrow-datafusion/issues/6195. We are trying to
1. keep the DataFusion code base clean and easy to contribute to and modify
2. Understand what features are covered and what are not

Part of doing so is having a single location for most test coverage so new tests can be added easily and people can follow the existing patterns easily

### Describe the solution you'd like

Port the tests in https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/sql/projection.rs to
`projection.slt` in https://github.com/apache/arrow-datafusion/tree/main/datafusion/sqllogictest/test_files

Note you can create catalogs and schema via SQL now

Catalogs: https://arrow.apache.org/datafusion/user-guide/sql/ddl.html#create-database
Schema: https://arrow.apache.org/datafusion/user-guide/sql/ddl.html#create-schema

Notes:
1. Instructions for running sqllogitest are here: https://github.com/apache/arrow-datafusion/tree/main/datafusion/sqllogictest

### Describe alternatives you've considered

_No response_

### Additional context

I think these are good first issues as they teach the contributor about the DataFusion codebase and tests, as well as see the end user apis in practice.

_No response_

### PR DESCRIPTION
**Pull request by PsiACE**
## Which issue does this PR close?

Closes #8210.

## Rationale for this change

Part of https://github.com/apache/arrow-datafusion/issues/6195.

## What changes are included in this PR?

Deleted projection.rs and added test cases in projection.slt.

## Are these changes tested?

Yes. I followed the conversion rules. 

Due to the fact that the projectioned data files used for testing are from other SQLLogicTest cases and do not exactly match the numbers generated by the original code, some of the test results may differ. However, I believe that they are all within expectations.

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/sqllogictest/test_files/projection.slt [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+SELECT V1.c1, v1.C2 FROM (SELECT test_simple.C1, TEST_SIMPLE.c2 FROM test_simple) V1 ORDER BY v1.c1, V1.C2 LIMIT 1;
+----
+0 0
+
+# projection on table scan
+query TT
+explain select c2 from test;
+----
+logical_plan TableScan: test projection=[c2]
+physical_plan CsvExec: file_groups={4 groups: [[WORKSPACE_ROOT/datafusion/core/tests/data/partitioned_csv/partition-0.csv], [WORKSPACE_ROOT/datafusion/core/tests/data/partitioned_csv/partition-1.csv], [WORKSPACE_ROOT/datafusion/core/tests/data/partitioned_csv/partition-2.csv], [WORKSPACE_ROOT/datafusion/core/tests/data/partitioned_csv/partition-3.csv]]}, projection=[c2], has_header=false
```
Comment (PsiACE) [inline datafusion/sqllogictest/test_files/projection.slt:156]: How to ignore the physical plan, it is unstable.

Comment (PsiACE) [reply datafusion/sqllogictest/test_files/projection.slt]: oh, we could:

```sql
set datafusion.explain.logical_plan_only = true
```

File: datafusion/core/tests/sql/partitioned_csv.rs [hunk 2]
```diff
@@ -19,30 +19,27 @@
...
 use tempfile::TempDir;
 
-/// Execute SQL and return results
-async fn plan_and_collect(
-    ctx: &mut SessionContext,
-    sql: &str,
-) -> Result<Vec<RecordBatch>> {
-    ctx.sql(sql).await?.collect().await
-}
+// /// Execute SQL and return results
```
Comment (alamb) [inline datafusion/core/tests/sql/partitioned_csv.rs:29]: Can you please just remove this code (rather than commenting it out)?

**Review by alamb — APPROVED** [review]
Thank you @PsiACE !

File: datafusion/core/tests/sql/partitioned_csv.rs (hunk 2 above)
Comment (PsiACE) [reply datafusion/core/tests/sql/partitioned_csv.rs]: In the latest commit, I removed them.

**Comment by PsiACE** [comment]
> Thank you @PsiACE !

I am glad that I have the opportunity to participate in the arrow community again in 2023. I think I will try to do some interesting things. It's also nice to meet you again.

**Comment by alamb** [comment]
> > Thank you @PsiACE !
> 
> I am glad that I have the opportunity to participate in the arrow community again in 2023. I think I will try to do some interesting things. It's also nice to meet you again.

Its great to see you back!

**Pull request merged**